src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/AS_RankSumTest.java

```diff
@@ -1,329 +1,121 @@
+        if (perAlleleValues.values().stream().allMatch(d -> d.isEmpty())){
+            return null;
         }
-    }
-
-    protected String makeRawAnnotationString(final List<Allele> vcAlleles, final Map<Allele, CompressedDataList<Integer>> perAlleleValues) {
-        String annotationString = "";
+        final StringBuilder annotationString = new StringBuilder();
         for (int i =0; i< vcAlleles.size(); i++) {
-            if (i!=0)
-                    annotationString += printDelim;
-            CompressedDataList<Integer> alleleValues = perAlleleValues.get(vcAlleles.get(i));
-            annotationString += alleleValues.toString();
-        }
-        return annotationString;
-    }
-
-    protected String makeReducedAnnotationString(VariantContext vc, Map<Allele,Double> perAltRankSumResults) {
-        String annotationString = "";
-        for (final Allele a : vc.getAlternateAlleles()) {
-            if (!annotationString.isEmpty())
-                annotationString += reducedDelim;
-            if (!perAltRankSumResults.containsKey(a))
-                logger.warn("ERROR: VC allele not found in annotation alleles -- maybe there was trimming?");
-            else
-                annotationString += String.format("%.3f", perAltRankSumResults.get(a));
+            if (i!=0) {
+                annotationString.append(printDelim);
+            }
+            final CompressedDataList<Integer> alleleValues = perAlleleValues.get(vcAlleles.get(i));
+            annotationString.append(alleleValues);
         }
-        return annotationString;
-    }
-
-    /**
-     *
-     * @param vc -- contains the final set of alleles, possibly subset by GenotypeGVCFs
-     * @param originalVC -- used to get all the alleles for all gVCFs
-     * @return
-     */
-    public  Map<String, Object> finalizeRawData(final VariantContext vc, final VariantContext originalVC) {
-        if (!vc.hasAttribute(getRawKeyName()))
-            return new HashMap<>();
-
-        final String rawRankSumData = vc.getAttributeAsString(getRawKeyName(),null);
-        if (rawRankSumData == null)
-            return new HashMap<>();
-
-        final Map<String,Object> annotations = new HashMap<>();
-        final AlleleSpecificAnnotationData<CompressedDataList<Integer>> myData = new AlleleSpecificAnnotationData(originalVC.getAlleles(), rawRankSumData);
-        parseRawDataString(myData);
-
-        final Map<Allele, Double> perAltRankSumResults = calculateReducedData(myData.getAttributeMap(), myData.getRefAllele());
-        //shortcut for no ref values
-        if (perAltRankSumResults.isEmpty())
-            return annotations;
-        final String annotationString = makeReducedAnnotationString(vc, perAltRankSumResults);
-        annotations.put(getKeyNames().get(0), annotationString);
-        return annotations;
+        return annotationString.toString();
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"})//FIXME generics here blow up
```

Comment:
Gross! Is there no way we can properly parameterize `ReducibleAnnotationData` in the method signature?